llmfoundry/data/finetuning/tasks.py

```diff
@@ -368,9 +366,46 @@ def build_from_hf(
         error: Optional[Exception] = None
         filtered_dataset = None
         try:
+            if safe_load:
+                if not os.path.isdir(dataset_name):
+                    # dataset_name is not a local dir path, download if needed.
+                    local_dataset_dir = os.path.join(
+                        _DOWNLOADED_FT_DATASETS_DIRPATH, dataset_name)
+
+                    if _is_empty_or_nonexistent(dirpath=local_dataset_dir):
+                        # Safely load a dataset from HF Hub with restricted file types.
+                        hf_hub.snapshot_download(
+                            dataset_name,
+                            repo_type='dataset',
+                            allow_patterns=[
+                                '*' + ext for ext in _SUPPORTED_EXTENSIONS
+                            ],
+                            token=hf_kwargs.get('token', None),
+                            local_dir_use_symlinks=False,
+                            local_dir=local_dataset_dir)
+                        if _is_empty_or_nonexistent(dirpath=local_dataset_dir):
+                            raise FileNotFoundError(
+                                f'safe_load is set to True. No data files with safe extensions {_SUPPORTED_EXTENSIONS} '
+                                + f'found for dataset {dataset_name}. ')
+                    # Set dataset_name to the downloaded location.
+                    dataset_name = local_dataset_dir
+
+                # dataset_name is a local dir path. Use the abspath to prevent confusion.
+                dataset_name = os.path.abspath(dataset_name)
+
+                # Ensure that the local dir contains only allowed file types.
+                dataset_files = [
```

Comment:
This check will always run even if dataset_name is a local folder. I think it's fine to have safe_load restrict to these file types even if it's a local folder, just wanted to point that out.